src/Controls/src/Build.Tasks/SetPropertiesVisitor.cs

```diff
@@ -719,7 +737,7 @@ static IEnumerable<Instruction> CompiledBindingGetSetter(TypeReference tSourceRe
 			setter.Body.InitLocals = true;
 
 			var il = setter.Body.GetILProcessor();
-			if (!properties.Any() || properties.Last().property.SetMethod == null)
+			if (!properties.Any() || properties.Last().property == null || properties.Last().property.SetMethod == null)
```

Comment:
You could simplify this to something like:
```suggestion
			if (properties.Last() is ThePropertyType p && p.property == null || p.property.SetMethod == null)
```
Which could avoid calling `Any()` and `Last()` as many times.

Reply:
you mean LastOrDefault ?